### PR TITLE
Chapter 3 staticpages

### PR DESCRIPTION
**Pull request by tenyiting**
ready

### CODE REVIEW

**Comment by tenyiting** [comment]
ready

**Comment by tenyiting** [comment]
ready

**Comment by tenyiting** [comment]
ready

**Comment by tenyiting** [comment]
Screenshot from 2021-08-24 17-28-41

**Comment by tenyiting** [comment]
ready

**Comment by tenyiting** [comment]
ready

File: app/assets/javascripts/posts.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
```
Comment (ngocvt-0484) [inline app/assets/javascripts/posts.coffee]: delete this file because not use

File: app/assets/javascripts/static_pages.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
```
Comment (ngocvt-0484) [inline app/assets/javascripts/static_pages.coffee]: delete this file because not use

File: app/assets/stylesheets/static_pages.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/static_pages.scss]: delete this file because not use

File: app/assets/stylesheets/posts.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Posts controller here.
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/posts.scss]: delete this file because not use

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
+base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb]: wrong indent 
image

File: app/helpers/posts_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module PostsHelper
```
Comment (ngocvt-0484) [inline app/helpers/posts_helper.rb]: delete this file because not use

File: app/helpers/static_pages_helper.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (ngocvt-0484) [inline app/helpers/static_pages_helper.rb]: delete this file because not use

File: app/views/layouts/_header.html.erb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
```
Comment (ngocvt-0484) [inline app/views/layouts/_header.html.erb:1]: format your html
it should be

Comment (ngocvt-0484) [reply app/views/layouts/_header.html.erb:1]: image

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -10,6 +10,10 @@
   </head>
 
   <body>
-    <%= yield %>
-  </body>
+<%= render "layouts/header" %>
+<div class="container">
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb]: format your html please

File: app/views/static_pages/help.html.erb [hunk 10]
```diff
@@ -0,0 +1,12 @@
+<%= provide :title, "Help" %>
+
+<h1>Help</h1>
+<p>
+Get help on the Ruby on Rails Tutorial at the
+<a href="http://www.railstutorial.org/help">Rails Tutorial
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: use link_to

File: test/controllers/posts_controller_test.rb [hunk 11]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class PostsControllerTest < ActionDispatch::IntegrationTest
```
Comment (ngocvt-0484) [inline test/controllers/posts_controller_test.rb]: dont need do test
please delete all file on test folder

File: app/assets/stylesheets/posts.scss (hunk 4 above)
Comment (tenyiting) [reply app/assets/stylesheets/posts.scss]: Screenshot from 2021-08-25 13-55-01
I hope this is correct. attached u with photo

File: app/helpers/posts_helper.rb (hunk 6 above)
Comment (tenyiting) [reply app/helpers/posts_helper.rb]: Screenshot from 2021-08-25 13-59-27

File: app/helpers/static_pages_helper.rb (hunk 7 above)
Comment (tenyiting) [reply app/helpers/static_pages_helper.rb]: Screenshot from 2021-08-25 14-01-45

File: test/controllers/posts_controller_test.rb (hunk 11 above)
Comment (tenyiting) [reply test/controllers/posts_controller_test.rb]: Screenshot from 2021-08-25 14-04-29

**Comment by tenyiting** [comment]
ready

File: app/controllers/posts_controller.rb [hunk 12]
```diff
@@ -0,0 +1,2 @@
+class PostsController < ApplicationController
```
Comment (ghost) [inline app/controllers/posts_controller.rb]: why you have PostsController?

File: app/models/post.rb [hunk 13]
```diff
@@ -0,0 +1,2 @@
+class Post < ApplicationRecord
```
Comment (ghost) [inline app/models/post.rb]: delete this file you don't need it

**Review by ghost — COMMENTED** [review]
why you still have 6 commits?

**Comment by tenyiting** [comment]
ready

**Comment by ngocvt-0484** [comment]
please change 6 commit to one commit

**Comment by ngocvt-0484** [comment]
you can using 
```
git rebase -i HEAD~6
```

File: test/models/user_test.rb [hunk 14]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
```
Comment (ngocvt-0484) [inline test/models/user_test.rb]: delete this file
you dont need do test

File: test/models/post_test.rb [hunk 15]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class PostTest < ActiveSupport::TestCase
```
Comment (ngocvt-0484) [inline test/models/post_test.rb]: delete this file
you dont need do test

File: test/fixtures/users.yml [hunk 16]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+one:
```
Comment (ngocvt-0484) [inline test/fixtures/users.yml]: delete this file
you dont need do test

File: test/fixtures/posts.yml [hunk 17]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+one:
```
Comment (ngocvt-0484) [inline test/fixtures/posts.yml]: delete this file
you dont need do test

File: test/controllers/static_pages_controller_test.rb [hunk 18]
```diff
@@ -0,0 +1,14 @@
+require 'test_helper'
+
+class StaticPagesControllerTest < ActionDispatch::IntegrationTest
```
Comment (ngocvt-0484) [inline test/controllers/static_pages_controller_test.rb]: delete this file
you dont need do test

File: app/views/static_pages/help.html.erb (hunk 10 above)
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: <%= link_to "Rails Tutorial help page", "http://www.railstutorial.org/help" %>

**Comment by tenyiting** [comment]
> you can using
> 
> ```
> git rebase -i HEAD~6
> ```
Screenshot from 2021-08-25 16-52-14

**Comment by tenyiting** [comment]
ready

**Comment by tenyiting** [comment]
> > you can using
> > ```
> > git rebase -i HEAD~6
> > ```
> 
> Screenshot from 2021-08-25 16-52-14

I changed in sublime after git rebase -i HEAD~6
pick..........
squash.........
squash..........
squash.......
.
.
.
.

**Comment by tenyiting** [comment]
ready

File: app/views/static_pages/help.html.erb [hunk 19]
```diff
@@ -0,0 +1,12 @@
+<%= provide :title, "Help" %>
+
+<h1>Help</h1>
+<p>
+  Get help on the Ruby on Rails Tutorial at the
```
Comment (tenyiting) [inline app/views/static_pages/help.html.erb]: now you can translate the text by using i18n then you can ready agin    https://phrase.com/blog/posts/i18n-a-simple-definition/

File: config/locales/en.yml [hunk 20]
```diff
@@ -1,33 +1,13 @@
...
-#   'true': 'foo'
-#
-# To learn more, please read the Rails Internationalization guide
-# available at http://guides.rubyonrails.org/i18n.html.
-
 en:
-  hello: "Hello world"
+  title: "Sample App"
+  p_1: "This is the home page for the"
+  p_2: "Ruby on Rails Tutorial"
```
Comment (tenyiting) [inline config/locales/en.yml:11]: translate languages..

**Comment by tenyiting** [comment]
ready

**Comment by tenyiting** [comment]
Ready

**Comment by tenyiting** [comment]
Ready

File: Gemfile [hunk 21]
```diff
@@ -15,9 +15,11 @@ gem 'sass-rails', '~> 5.0'
 gem 'uglifier', '>= 1.3.0'
 # See https://github.com/rails/execjs#readme for more supported runtimes
 # gem 'mini_racer', platforms: :ruby
-
-# Use CoffeeScript for .coffee assets and views
-gem 'coffee-rails', '~> 4.2'
+Gemfile
```
Comment (ngocvt-0484) [inline Gemfile]: ???
why you add it?
please delete

File: Gemfile [hunk 22]
```diff
@@ -15,9 +15,11 @@ gem 'sass-rails', '~> 5.0'
...
+app/helpers/application_helper.rb
```
Comment (ngocvt-0484) [inline Gemfile]: ???
why you add it?
please delete

File: Gemfile [hunk 23]
```diff
@@ -15,9 +15,11 @@ gem 'sass-rails', '~> 5.0'
...
+app/views/layouts/application.html.erb
+config/locales/en.yml
+config/routes.rb
 # Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
```
Comment (ngocvt-0484) [inline Gemfile]: same

File: test/models/user_test.rb (hunk 14 above)
Comment (ngocvt-0484) [reply test/models/user_test.rb]: not fix yet

**Comment by tenyiting** [comment]
ready

**Review by ghost — COMMENTED** [review]
You have 2 commits, please combine into 1 
Some file test you didn't delete, please delete it

File: test/models/post_test.rb (hunk 15 above)
Comment (tenyiting) [reply test/models/post_test.rb]: Screenshot from 2021-08-30 20-19-04

File: test/fixtures/users.yml (hunk 16 above)
Comment (tenyiting) [reply test/fixtures/users.yml]: Screenshot from 2021-08-30 20-21-50

**Comment by tenyiting** [comment]
ready

**Review by thaodtd-0665 — APPROVED** [review]

**Comment by thaodtd-0665** [comment]
merged

**Pull request merged**